StyleChecker/src/analyzerTypes/AnalyzerType.java

```
@@ -0,0 +1,8 @@
+package analyzerTypes;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+public interface AnalyzerType {
+    void analyze(String line, BufferedWriter output) throws IOException;
```

Why does the analyzer knows where to write the output? It would be better to return the error string and then the consumer of the analyzer knows how to "print" the error.